src/GraphWalker.cpp

```diff
@@ -39,7 +39,7 @@ static bool validConnection(Structure* src, Structure* dst, Direction _d)
 	if (src == nullptr || dst == nullptr)
 	{
 		#ifdef DEBUG
-		throw std::runtime_error("GraphWalker::validConnection() was passed a NULL Pointer.");
+		std::cout << "GraphWalker::validConnection() was passed a NULL Pointer." << std::endl;

```

Comment:
I don't see why this change is included in this change set.

I would be tempted to argue that a `nullptr` value there is a programming error that should crash hard. Though perhaps a better design would use references instead, which are not nullable.

Reply:
Debug builds complain of unreachable code, otherwise. To clarify: if you're checking for `nullptr`, by definition a `nullptr` value isn't exceptional and shouldn't throw.